### PR TITLE
Implement tier 1 MCG replication policy tests

### PR DESCRIPTION
**Pull request by MeridianExplorer**

### CODE REVIEW

File: ocs_ci/ocs/bucket_utils.py [hunk 1]
```diff
@@ -1307,3 +1307,73 @@ def get_bucket_available_size(mcg_obj, bucket_name):
...
+            )
+            return False
+
+    try:
+        for comparison_result in TimeoutSampler(600, 30, _comparison_logic):
+            if comparison_result:
+                return True
+    except TimeoutExpiredError:
+        logger.error("The compared buckets do not contain the same set of objects")
+        assert False, "The compared buckets do not contain the same set of objects"
```
Comment (ebenahar) [inline ocs_ci/ocs/bucket_utils.py:1345]: IMO, it makes more sense to return bool here and handle the return value in the test by raising an exception if needed

File: ocs_ci/ocs/bucket_utils.py [hunk 2]
```diff
@@ -1307,3 +1307,73 @@ def get_bucket_available_size(mcg_obj, bucket_name):
...
+    Upload random objects to a bucket
+    """
+    full_object_path = f"s3://{bucket_to_write}"
+    object_list = []
+
+    for i in range(amount):
+        file_name = f"testfile{i}.txt"
+        object_list.append(file_name)
+        awscli_pod.exec_cmd_on_pod(
+            f"dd if=/dev/urandom of={original_dir}/{file_name} bs=1M count=1 status=none"
```
Comment (ebenahar) [inline ocs_ci/ocs/bucket_utils.py:1366]: we have the same command in ocs_ci.ocs.bucket_utils.setup_base_objects.
Please save it as a constant and call it from all places

File: ocs_ci/ocs/resources/mcg.py [hunk 3]
```diff
@@ -770,13 +789,28 @@ def cli_create_bucketclass(
...
+            replication_policy_file.write(
+                json.dumps(replication_policy).encode("utf-8")
+            )
+            replication_policy = (
+                f" --replication-policy {replication_policy_file.name}"
+                if replication_policy
+                else ""
+            )
+            self.exec_mcg_cmd(
+                f"bucketclass create {placement_type}{name}{backingstores}{placement_policy}{replication_policy}"
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:812]: is it possible only with MCG CLI?

File: ocs_ci/ocs/bucket_utils.py (hunk 1 above)
Comment (MeridianExplorer) [reply ocs_ci/ocs/bucket_utils.py:1345]: Addressed

**Comment by MeridianExplorer** [comment]
[PR verified](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/6961/) (ran a deployment job instead of a PR validation one by mistake)

File: tests/manage/mcg/test_bucket_replication.py [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+        argnames=["source_bucketclass", "target_bucketclass"],
+        argvalues=[
+            pytest.param(
+                {
+                    "interface": "OC",
+                    "backingstore_dict": {"aws": [(1, "eu-central-1")]},
+                },
+                {"interface": "OC", "backingstore_dict": {"azure": [(1, None)]}},
+                # TODO: add polarion id
+                marks=[tier1, pytest.mark.polarion_id()],  # TODO
```
Comment (ebenahar) [inline tests/manage/mcg/test_bucket_replication.py:39]: polarion ids are missing

**Review by ebenahar — DISMISSED** [review]

**Review by bernerhat — COMMENTED** [review]
Overall very nice work!

I'm not sure i am following correctly, are you accepting only one replication rule per entity? (bc/bucket) i believe we should allow multiple rules for a scenario such as replicating to two different buckets for different object prefix

File: ocs_ci/ocs/bucket_utils.py [hunk 5]
```diff
@@ -1019,6 +1019,35 @@ def namespace_bucket_update(mcg_obj, bucket_name, read_resource, write_resource)
     )
 
 
+def write_random_objects(io_pod, file_dir, amount, pattern="ObjKey"):
```
[the remark this answers is not in the document]
Comment (MeridianExplorer) [reply ocs_ci/ocs/bucket_utils.py]: #5050

**Comment by MeridianExplorer** [comment]
@bernerhat - Submitted #5051.
At the moment I am leaving this out since I believe it's out of scope for tier1, and we should merge it first before expanding into more rules or more complex rules, such as ones with prefixes (which we also do not test automatically at the moment)

**Review by bernerhat — APPROVED** [review]
Thanks for opening the issues :)
LGTM

**Review by ebenahar — APPROVED** [review]

**Pull request merged**